pkg/operatorsource/datastore.go

```diff
@@ -0,0 +1,54 @@
+package operatorsource
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/operator-framework/operator-marketplace/pkg/appregistry"
+)
+
+var (
+	ErrManifestNotFound = errors.New("manifest not found")
+)
+
+type DatastoreReader interface {
+	// Read returns the associated operator manifest given a package ID
+	Read(packageID string) (*appregistry.OperatorMetadata, error)
+}
+
+func newDatastore() *hashmapDatastore {
+	return &hashmapDatastore{
+		list: map[string]*appregistry.OperatorMetadata{},
+	}
+}
+
+type hashmapDatastore struct {
+	list map[string]*appregistry.OperatorMetadata
```

Comment:
Change list to something appropriate.

Reply:
hmm, I can rename it to `packages`. That is not so generic as `list`